### DISCUSSION TITLE
A new way to Search 🔎  and Navigate code - Post Universe Check-in

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#39163] [Other Feature Feedback, Questions, & Ideas]
Hi everyone 👋, Universe 2022 🛰️ was beyond exciting and I still can’t get over the amazing new additions to the Code Search and Navigation products. 💯  During Universe we introduced a brand new search and code navigation view and it is loaded with new features. Let’s get into it! ⏭️  

## Search more effectively 🆕 

Our engineers have completely redesigned and rebuilt this experience from the ground up adding powerful capabilities like regular expressions, Boolean expressions, qualifiers, symbol search, and more! So, you can now save a lot of time and find code effortlessly.

![Screenshot 2022-11-10 at 6 14 32 PM](https://user-images.githubusercontent.com/95834130/201676977-36a41ebe-99c5-4cfa-a821-de7566063c13.png)

These improvements replace the 2021 [technology preview](https://github.blog/2021-12-08-improving-github-code-search/) for GitHub code search at [cs.github.com](http://cs.github.com/).

##  Symbols inspector 👮 

Understanding how a symbol is being used inside a repository or file is hard, especially in files you don’t contribute to. Now, you can just click any code symbol and get an instant overview of how it’s used throughout the repository.

![Screenshot 2022-11-10 at 6 42 56 PM](https://user-images.githubusercontent.com/95834130/201677059-2508336b-6813-4548-9f14-3cad9070c147.png)

## An all-new search and browsing experience 🏅

1. The new incredible search engine is focused on code as it can find relevant results with incredible speed. So, save time and find code directly inside a repository with just a few keystrokes ⌨️  and jump to it without having to visit search results first. 

![Screenshot 2022-11-10 at 6 49 59 PM](https://user-images.githubusercontent.com/95834130/201677156-00d3a0b9-717b-4e03-8df8-2d4e882ac5bd.png)

We've designed this code viewing experience to provide a generational jump in code browsing and viewing on GitHub and to make it more accessible. It has some amazing add-ons such as the new tree pane on the left and the symbols pane on the right side, though there is more than just that. For more features, read the [changelog](https://github.blog/changelog/2022-11-09-introducing-an-all-new-code-search-and-code-browsing-experience/#a-better-way-to-search-code). 💟 

![Screenshot 2022-11-10 at 7 03 42 PM](https://user-images.githubusercontent.com/95834130/201677310-cf23d603-3838-49ce-b744-1c407ce1419a.png)

2. Code navigation features ‘jump to definition’ and ‘find all references’ are now available for all Rust projects on GitHub. See more [here](https://github.blog/changelog/2022-11-09-search-based-code-navigation-for-rust-projects/).

3. We’ve also shipped 🚢   the [Cross-repo code navigation](https://github.blog/changelog/2022-11-09-cross-repo-code-navigation-for-python-projects/) for all Python 🐍 repositories. When showing the definition of a function or method, we now include definitions from other repositories, and from the Python standard library.

We are eager for you to try the 🆕  code search and code view beta! [Join the waitlist](https://github.com/features/code-search-code-view/signup) to get access.

This project is a major update to GitHub's user experience that was made possible by the feedback you provided❤️.  Help make the experience even better by [sharing your latest feedback here](https://github.com/orgs/community/discussions/38692).

Got any questions about the new features? Jump on to this [discussion](https://github.com/orgs/community/discussions/38692) in the community and ride along with other members or [post a new discussion](https://github.com/community/community/discussions/new?category=code-search-and-navigation). Don’t forget to drop in your new favorite feature shipped this Universe 🚀

**Comment by david-engelmann** [comment]
Phenomenal!

**Comment by tonglil** [comment]
The find feature while viewing a file is the only part of this change that makes me want to disable the entire thing.

**Please do not hijack the user's `cmd-f`/`ctrl-f`!** There are so many things your finder doesn't do that my browser does:
- match case
- match diacritics
- whole word matching

Additionally:
1. it ruins the browser history with every next/prev match, resulting in me having to press back/`cmd-[` multiple times.
2. it's a jarring experience compared to my find usage on every other page on the internet (internal docs, other GH pages, google, etc). GitHub is not a "standalone" app like Slack nor is it so self contained like Google Docs, so **it should not take over find by default**.
3. the `cmd-f` feature does not work the same across GH: when editing code in the web UI, hitting "enter" after finding does not go to the next instance - instead it replaces the found word with a new line (you should improve that before you touch code view `cmd-f`).
4. I lose find when I want to see what I found and the symbols list.

Having to hit the shortcut twice is very annoying. To propose some alternatives:
1. hit it twice to use Github's finder/find menu.
2. let me disable find separately from the rest of the new UI.

From the UX, there are lots of instances where people do not like this behavior:
- https://meta.discourse.org/t/options-to-disable-hijack-of-cmd-f-ctrl-f-and-keys-for-search/16875/50
- https://old.reddit.com/r/assholedesign/comments/5ydmcd/captures_ctrlf_to_open_their_own_search_function/
- https://superuser.com/questions/1038172/chrome-search-in-page-when-ctrlf-shortcut-is-hijacked

Thankfully in searching for a solution to this new problem, I found `cmd-g` (hopefully GH doesn't decide to hijack that too).

**Comment by 0xdevalias** [comment]
I would also like to be able to paste a full file path, with optional line number into the cmd+k dialog, and have it also suggest opening that file, at that line number.

> I would like to be able to use the `t` (Jump to file) shortcut, and give it a full path, optionally also with a line number, such as `foo/bar/baz/index.js:123` and have it jump to the correct file and line.
> 
> I believe it used to be possible to use this with the file path (without line number), but now even that seems to fail using the new code search/etc beta.
> 
> _Originally posted by @0xdevalias in https://github.com/orgs/community/discussions/9671#discussioncomment-4385739_

**Reply by 0xdevalias** [reply]
See also: https://github.com/orgs/community/discussions/38692#discussioncomment-4100798

**Comment by Akash1134** [comment]
Hi All, thanks for sharing your valuable feedback. While it is super-important for us to hear your feedback, it's also important to have it consolidated with other product feedback.  Please post all feedback and feature requests to this [discussion](https://github.com/orgs/community/discussions/38692) so our product team can track feedback all in one place.

**Reply by 0xdevalias** [reply]
@Akash1134 FYI seems to be a broken link `https://github.com/orgs/community/discussions/38692https://github.com/orgs/community/discussions/38692`

Correct working link:

- https://github.com/orgs/community/discussions/38692

**Comment by ghost** [comment]
Hi everyone. +1 great idea.

**Comment by prabhasjoshi** [comment]
How to de-activate this symbol search ?
https://stackoverflow.com/questions/76338535/how-to-stop-github-symbol-search-modals-on-click
